src/pages/index.js

```diff
@@ -126,47 +131,43 @@ const cardPreview = new PopupWithImage({
 const newCardPopup = new PopupWithForm({
   popupSelector: "#add-modal",
   handleFormSubmit: (inputValues) => {
-    api.addCard(inputValues.name, inputValues.link).then((response) => {
-      renderCard(response);
-    });
-    newCardPopup.close();
+    newCardPopup.renderLoading(true);
+    api
+      .addCard(inputValues.name, inputValues.link)
+      .then((response) => {
+        newCardPopup.renderLoading(false);
```

Comment:
`.renderLoading(false)`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.